dev/PushNotifications/PushNotificationActivationInfo.cpp

```diff
@@ -0,0 +1,34 @@
+﻿#include "pch.h"
+#include "PushNotificationActivationInfo.h"
+#include "Microsoft.Windows.PushNotifications.PushNotificationActivationInfo.g.cpp"
+
+#include <Windows.ApplicationModel.background.h>
+
+namespace winrt::Microsoft::Windows::PushNotifications::implementation
+{
+    PushNotificationActivationInfo::PushNotificationActivationInfo(Microsoft::Windows::PushNotifications::PushNotificationRegistrationKind const& kind, winrt::guid const& taskClsid)
+    {
+        m_kind = kind;
+        m_taskClsid = taskClsid;
+    }
+    winrt::guid PushNotificationActivationInfo::TaskClsid()
+    {
+        auto lock = m_lock.lock_shared();
+        return m_taskClsid;
+    }
+    Microsoft::Windows::PushNotifications::PushNotificationRegistrationKind PushNotificationActivationInfo::Kind()
+    {
+        auto lock = m_lock.lock_shared();
+        return m_kind;
+    }
+    winrt::com_array<winrt::Windows::ApplicationModel::Background::IBackgroundCondition> PushNotificationActivationInfo::GetConditions()
+    {
+        auto lock = m_lock.lock_shared();
+        return winrt::com_array<winrt::Windows::ApplicationModel::Background::IBackgroundCondition>(m_backgroundConditions.begin(), m_backgroundConditions.end());
+    }
+    void PushNotificationActivationInfo::SetConditions(array_view<winrt::Windows::ApplicationModel::Background::IBackgroundCondition const> conditions)
+    {
+        auto lock = m_lock.lock_exclusive();
+        m_backgroundConditions = winrt::com_array<winrt::Windows::ApplicationModel::Background::IBackgroundCondition>(conditions.begin(), conditions.end());
```

Comment:
Nit: we can improve readability a little bit by declaring `using namespace`.

Reply:
One trick that you can use is define namespaces like this:
namespace winrt
{
    using namespace Windows::Foundation;
    using namespace Windows::Foundation::Collections;
    using namespace Windows::Networking::PushNotifications;
}

Then you can infer to constructs in the namespace like this:
winrt::PushNotificationChannelManager channelManager{}

This way, there is no confusion as to which version of the PushNotificationChannelManager is being inferred to: ie. the c++ winrt projection in this case Vs the ABI